delta/rdf/src/test/scala/ch/epfl/bluebrain/nexus/delta/rdf/jsonld/ExpandedJsonLdSpec.scala

```diff
@@ -142,20 +150,36 @@ class ExpandedJsonLdSpec extends AnyWordSpecLike with Matchers with Fixtures {
 
     "remove a key" in {
       val multiRoot = jsonContentOf("/jsonld/expanded/input-multiple-roots.json")
-      val batmanIri = iri"http://example.com/batman"
-      val name      = iri"http://example.com/name"
-      val json      = json"""[{"@id": "$iri"}, {"@id": "$batmanIri", "$name": [{"@value": "Batman"} ] }]"""
+      val batmanIri = iri"$example/batman"
+      val name      = iri"$example/name"
+      val json      =
+        json"""[{"@id": "$iri", "@type": ["$example/Person"]}, {"@id": "$batmanIri", "@type": ["$example/Person", "$example/Hero"], "$name": [{"@value": "Batman"} ] }]"""
 
       ExpandedJsonLd(multiRoot).accepted.remove(name) shouldEqual ExpandedJsonLd.expanded(json).rightValue
     }
 
     "remove a key from all entries" in {
       val multiRoot = jsonContentOf("/jsonld/expanded/input-multiple-roots.json")
-      val batmanIri = iri"http://example.com/batman"
-      val name      = iri"http://example.com/name"
-      val json      = json"""[{"@id": "$iri"}, {"@id": "$batmanIri" }]"""
+      val batmanIri = iri"$example/batman"
+      val name      = iri"$example/name"
+      val json      =
+        json"""[{"@id": "$iri", "@type": ["$example/Person"]}, {"@id": "$batmanIri", "@type": ["$example/Person", "$example/Hero"] }]"""
 
       ExpandedJsonLd(multiRoot).accepted.removeFromEntries(name) shouldEqual ExpandedJsonLd.expanded(json).rightValue
     }
+
+    "filter type" in {
+      val multiRoot = jsonContentOf("/jsonld/expanded/input-multiple-roots.json")
+      val batmanIri = iri"$example/batman"
+      val name      = iri"$example/name"
+      val json      =
+        json"""[{"@id": "$batmanIri", "@type": ["$example/Person", "$example/Hero"], "$name": [{"@value": "Batman"}] }]"""
+
+      ExpandedJsonLd(multiRoot).accepted.filterType(iri"$example/Hero") shouldEqual
+        ExpandedJsonLd.expanded(json).rightValue
+
+      ExpandedJsonLd(multiRoot).accepted.filterType(iri"$example/Other") shouldEqual
```

Comment:
I would also add a filterType test with `iri"$example/Person"` to check that it echoes back the entire document.